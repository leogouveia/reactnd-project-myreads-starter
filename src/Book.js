import React, { Component } from 'react';
import PropTypes from 'prop-types';
import BookshelfChanger from './BookshelfChanger';

export class Book extends Component {
  state = {};

  render() {
    const { titleName, authors, coverImage } = this.props;
    return (
      <div className="book">
        <div className="book-top">
          <div
            className="book-cover"
            style={{
              width: 128,
              height: 193,
              backgroundImage: `url(${coverImage})`,
            }}
          />
          <BookshelfChanger />
        </div>
        <div className="book-title">{titleName}</div>
        <div className="book-authors">{authors.concat(' ')}</div>
      </div>
    );
  }
}

Book.propTypes = {
  titleName: PropTypes.string.isRequired,
  authors: PropTypes.array.isRequired,
  coverImage: PropTypes.string.isRequired,
};
export default Book;
