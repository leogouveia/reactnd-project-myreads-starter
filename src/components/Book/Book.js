import React, { Component } from "react";
import PropTypes from "prop-types";
import _ from "lodash";
import "./Book.css";
import noCoverImage from "./no-cover.jpg"

export class Book extends Component {
  state = {};

  render() {
    const { book, children } = this.props;
    const { title, authors, thumbnail } = this.getBookData(book);

    return (
      <div className="book">
        <div className="book-top">
          <div
            className="book-cover"
            style={{
              width: 128,
              height: 193,
              backgroundImage: `url(${thumbnail || noCoverImage})`
            }}
          />
          {children}
        </div>
        <div className="book-title">{title}</div>
        <div className="book-authors">{authors}</div>
      </div>
    );
  }

  getBookData(book) {
    let { title, authors, imageLinks } = book;
    title = title || "";
    authors = !Array.isArray(authors) ? authors : authors.join(", ");
    const thumbnail =
      imageLinks && _.has(imageLinks, "thumbnail") ? imageLinks.thumbnail : "";
    return { title, authors, thumbnail };
  }
}

Book.propTypes = {
  book: PropTypes.object.isRequired,
  children: PropTypes.node
};

Book.defaultProps = {
  authors: []
};

export default Book;
