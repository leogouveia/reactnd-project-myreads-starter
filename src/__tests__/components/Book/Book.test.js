import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { Book } from "../../../components/Book/Book";

describe("<Book />", () => {
  it("should render a book", () => {
    const bookData = {
      title: "Book Title",
      authors: "Jonh Doe",
      imageLinks: { thumbnail: "http://fakeimage.com/image.png" }
    };
    const wrapper = shallow(<Book book={bookData} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should render a book without authors and cover", () => {
    const bookData = { title: "Book Title" };
    const wrapper = shallow(<Book book={bookData} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
