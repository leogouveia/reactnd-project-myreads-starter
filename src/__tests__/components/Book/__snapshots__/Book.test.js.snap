// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Book /> should render a book 1`] = `
<div
  className="book"
>
  <div
    className="book-top"
  >
    <div
      className="book-cover"
      style={
        Object {
          "backgroundImage": "url(http://fakeimage.com/image.png)",
          "height": 193,
          "width": 128,
        }
      }
    />
  </div>
  <div
    className="book-title"
  >
    Book Title
  </div>
  <div
    className="book-authors"
  >
    Jonh Doe
  </div>
</div>
`;

exports[`<Book /> should render a book without authors and cover 1`] = `
<div
  className="book"
>
  <div
    className="book-top"
  >
    <div
      className="book-cover"
      style={
        Object {
          "backgroundImage": "url()",
          "height": 193,
          "width": 128,
        }
      }
    />
  </div>
  <div
    className="book-title"
  >
    Book Title
  </div>
  <div
    className="book-authors"
  />
</div>
`;
